### DISCUSSION TITLE
Is there a way to export the results consult session to an occur/compile/grep-style buffer?

### DISCUSSION THREAD

**Discussion by cperivol** [minad/consult#898] [Q&A] [answered]
I like consult a lot but I often find myself in a situation where I want to do something complicated with each one of the results I painstakingly narrowed my session down to. To motivate let me provide 3 use-cases:

- I have all the files/lines that I want to refactor one-by-one in the session which is the job for an occur buffer but not so much for the completion list. 
- Sometimes I don't know exactly what I am looking for and I want to look at a larger portion of the results at a time than the 5-6 results displayed in the minibuffer.
- I want to do textual manipulation on the result list, eg I want to collect all the values a config variable takes in a list.

Thanks!

EDIT: Upon careful consideration I think different consult sources are better suited for different modes so the *absolute ideal* (but unlikely to be already true) scenario would be to have `consult-grep` to pop up a `grep-mode` buffer, `consult-buffer` to `ibuffer`, etc. Has anyone tried something like this?

**Comment by minad** [comment, marked as answer]
Yes, this functionality is provided by the Embark package. Check out the Consult and Embark README for further information.
